x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/history/HistoryStore.java

```diff
@@ -31,17 +32,19 @@
     private static final Logger logger = LogManager.getLogger(HistoryStore.class);
 
     private final BulkProcessor bulkProcessor;
+    private final ClusterService clusterService;
 
-    public HistoryStore(BulkProcessor bulkProcessor) {
+    public HistoryStore(BulkProcessor bulkProcessor, ClusterService clusterService) {
```

Comment:
Do you think it might be better to change this to a `Supplier<ClusterState>` so that it's clearer that we just need to access the current `ClusterState`? That would let us simplify the mocking a bit and make it clearer that we're not e.g. adding listeners here.

Reply:
[done](https://github.com/elastic/elasticsearch/pull/56734/commits/123453ccfff325b6fa364944dd19a37bc7e789c2)